mods/ra/maps/allies-04/allies04-AI.lua

```diff
@@ -0,0 +1,264 @@
+	DefendActor(unit)
+	Trigger.OnKilled(unit, function() HarvesterKilled = true end)
+end
+
+DefendActor = function(unit)
+	Trigger.OnDamaged(unit, function(self, attacker)
+		if AttackOnGoing then
+			return
+		end
+		AttackOnGoing = true
+
+		local Guards = SetupAttackGroup()
+
+		if #Guards <= 0 then
+			AttackOnGoing = false
+			return
+		end
+
+		Utils.Do(Guards, function(unit)
+			if not self.IsDead then
+				unit.AttackMove(self.Location)
+			end
+			IdleHunt(unit)
+		end)
+
+		Trigger.OnAllRemovedFromWorld(Guards, function() AttackOnGoing = false end)
+	end)
+end
+
+InitAIUnits = function()
+	IdlingUnits = ussr.GetGroundAttackers()
+
+	DefendActor(Conyard)
+	for i,v in ipairs(InitialBase) do
+		DefendActor(v)
+		Trigger.OnDamaged(v, function(building)
+			if building.Owner == ussr and building.Health < building.MaxHealth * 3/4 then
+				building.StartBuildingRepairs()
+			end
+		end)
+
+		Trigger.OnKilled(v, function()
+			BaseBuildings[i].exists = false
+		end)
+	end
+end
+
+ProduceInfantry = function()
+	if HoldProduction then
+		Trigger.AfterDelay(DateTime.Minutes(1), ProduceInfantry)
+		return
+	end
+
+	-- See AttackDelay in WorldLoaded
+	local delay = Utils.RandomInteger(AttackDelay[1], AttackDelay[2])
+	local toBuild = { Utils.Random(SovietInfantryTypes) }
+	ussr.Build(toBuild, function(unit)
+		IdlingUnits[#IdlingUnits + 1] = unit[1]
+		Trigger.AfterDelay(delay, ProduceInfantry)
+
+		-- See AttackGroupSize in WorldLoaded
+		if #IdlingUnits >= (AttackGroupSize * 2.5) then
+			SendAttack()
+		end
+	end)
+end
+
+ProduceVehicles = function()
+	if HoldProduction then
+		Trigger.AfterDelay(DateTime.Minutes(1), ProduceVehicles)
+		return
+	end
+
+	-- See AttackDelay in WorldLoaded
+	local delay = Utils.RandomInteger(AttackDelay[1], AttackDelay[2])
+	if HarvesterKilled then
+		ussr.Build({ "harv" }, function(harv)
+			ProtectHarvester(harv[1])
+			HarvesterKilled = false
+			Trigger.AfterDelay(delay, ProduceVehicles)
+		end)
+	else
+		local toBuild = { Utils.Random(SovietVehicleTypes) }
+		ussr.Build(toBuild, function(unit)
+			IdlingUnits[#IdlingUnits + 1] = unit[1]
+			Trigger.AfterDelay(delay, ProduceVehicles)
+
+			-- See AttackGroupSize in WorldLoaded
+			if #IdlingUnits >= (AttackGroupSize * 2.5) then
+				SendAttack()
+			end
+		end)
+	end
+end
+
+ProduceAircraft = function()
+	ussr.Build(SovietAircraftType, function(units)
+		local yak = units[1]
+
+		Trigger.OnKilled(yak, ProduceAircraft)
+
+		-- Declare the variables outside the OnIdle Trigger
+		local target
+		Trigger.OnIdle(yak, function()
+			if not target or (not target.IsInWorld) then
+
+				local enemies = Utils.Where(Map.ActorsInWorld, function(self) return self.Owner == player and self.HasProperty("Health") end)
+				if #enemies > 0 then
+					target = Utils.Random(enemies)
+				else
+					yak.Wait(DateTime.Seconds(5))
+				end
+			end
+
+			if target and yak.AmmoCount() > 0 then
+				yak.Attack(target)
+			else
+				yak.ReturnToBase()
+				yak.Resupply()
```

Comment:
The Yak gets stuck in a `WaitFor` activity here. I'm not quite sure, which one. Probably it's [this one](https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Activities/Air/ResupplyAircraft.cs#L39), but simply adding a `yak.Wait(1)` should have fixed that then, which it didn't.